gemsfx/src/main/java/com/dlsc/gemsfx/skins/SVGImageViewSkin.java

```diff
@@ -36,6 +37,36 @@ public SVGImageViewSkin(SVGImageView svgImageView) {
 
         loadingImage();
         getChildren().add(imageView);
+
+        registerWindowScaleListeners(svgImageView);
+    }
+
+    /**
+     * Registers listeners on the current Scene and its Window for render-scale changes.
+     * <p>
+     * This ensures that when the {@link Scene} or its {@link javafx.stage.Window} changes (e.g., moving
+     * from a normal DPI display to a high DPI/4K display), the renderScaleX and renderScaleY
+     * property listeners will be added or removed appropriately.
+     *
+     * @param svgImageView the SVGImageView whose Scene and Window changes need to be tracked.
+     */
+    private void registerWindowScaleListeners(SVGImageView svgImageView) {
+        Scene scene = svgImageView.getScene();
+        if (scene != null && scene.getWindow() != null) {
+            scene.getWindow().renderScaleXProperty().addListener(weakListener);
+            scene.getWindow().renderScaleYProperty().addListener(weakListener);
+        }
+
+        svgImageView.sceneProperty().addListener((obs, oldScene, newScene) -> {
+            if (oldScene != null && oldScene.getWindow() != null) {
```

Comment:
But what if there is no window? Then this will never be called again after the scene has been assigned? What about using the TreeShowing utility class from Pro?